content/developer/reference/guidelines.rst

```diff
@@ -999,17 +999,88 @@ More precise JS guidelines are detailed in the `github wiki  <https://github.com
 You may also have a look at existing API in Javascript by looking Javascript
 References.
 
-CSS coding guidelines
----------------------
+Template guidelines
+-------------------
+
+Structure
+~~~~~~~~~
+
+.. note:: These rules are in place because the general structure of the template is considered more important than the details of each line.
+
+- Each block element should be on its own line.
+- Small inline/style/text elements can be placed either on the same line as their parent or on their own line (depending on their importance and size).
+- Indentation of 4 spaces is applied between a parent and its children elements. The closing tag of a parent should always be at the same indentation level as its opening tag.
+- It is preferable to use instructions (such as `t-if`) directly on the element on which they apply (as opposed to having them on a separate `<t>` element).
+
+Naming
+~~~~~~
+
+The root element and each other *important* element of the template should have a `name` attribute to ease the readability of the template.
+
+.. note::
+    The importance of an element in regard to this rule is left to the appreciation of the reviewer. If the structure of the template is hard to follow when reading just the first words of each line, it is probably a sign that some `name` attributes are missing on those lines.
+
+The name on the root element should follow the format `o-ComponentName`. And for children elements `o-ComponentName-partName` where:
+
+- `o-` is a must have Odoo prefix.
+- `ComponentName` is the PascalCase name of the current component/template.
+- `-` is a mandatory hyphen to separate the component name and the part name.
+- `partName` is the camelCase name of the child element referred by this `name` attribute.
+
+Attributes
+~~~~~~~~~~
+
+All the attributes of an element should be on the same line as the element itself. No character limit per line.
```

Comment:
not a fan. it's much easier to review the content of a diff if I can see what changed precisely, and it's easier to see that if attributes/props are on several lines. But since it is left to the appreciation of the reviewer, I guess I don't care either

Reply:
This kind of guideline usually goes the other way around: make code clearer and more easily reviewable at the expense of the number of lines.

Funnily enough, you didn't follow this repository's guidelines regarding the character limit per line in this PR.

Reply:
@AntoineVDV thank you for the link, I didn't know this page existed. I'll make a pass to apply those once we are ok with the content.

@ged-odoo I agree with you here, it's a tradeoff between ease of diff and better overview of the template.
But both can depend on tooling so it's not that important either way if/once we have proper tools.

Same as the other point, as this one is left to appreciation of reviewer, it doesn't really matter.

Reply:
the question is how much value we get out of both representation as we write it. Obviously, they both have the same xml structure, but they convey the information in a different way. I actually think that multiple lines are easier to review and understand. I spend much more time tracking a particular attribute/prop than trying to understand the structure of a template. So in this case, I think this should be removed from the guidelines